### PR TITLE
Lab 05

### PR DESCRIPTION
**Pull request by bodnarira**

### CODE REVIEW

File: ISP.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class IRect:
```
Comment (polyakovartem) [inline ISP.py]: I don't see interface segregation here

File: LSP.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+class Lien(Animal):
+    def legs(self):
+        return 4
+
+class Monkey(Animal):
+    def legs(self):
+        return 2
+
+def GetLegs (animal: Animal):
+    car.legs()
```
Comment (polyakovartem) [inline LSP.py]: where did you copy/paste this from? :)

**Review by polyakovartem — CHANGES_REQUESTED** [review]
please provide usage example, some of your code will not even work. How did you test it?

**Comment by polyakovartem** [comment]
please create more decent examples

File: LSP.py (hunk 2 above)
Comment (bodnarira) [reply LSP.py]: Ohhh, sorry.
I have made a mistake.
At first I made example with cars but then my friend told me that my example is wrong and advised me to do example with animal legs. And I forgot to change car to animal.

**Review by bodnarira — COMMENTED** [review]
More decent for which exactly examples?

**Review by polyakovartem — APPROVED** [review]
ISP is missing

**Review by polyakovartem — COMMENTED** [review]
the word "segregation" implies that at least two entities exist to segregate

**Pull request closed without merging**